test/app-luatest/gh_4317_console_backslash_test.lua

```diff
@@ -0,0 +1,63 @@
+local t = require('luatest')
+local g = t.group()
+local console = require('console')
+local socket = require('socket')
+local log = require('log')
+
+local result_str =  [[tarantool> local a = 0 \
+         > for i = 1, 10 do
+         > a = a + i
+         > end \
+         > print(a)
+55
+---
+...
+
+tarantool> ]]
+
+local build_dir = arg[-1]:gsub('tarantool$', '')
+
+g.test_local_console = function()
+    t.skip_if(jit.os == 'OSX', "LD_PRELOAD doesn't work on OSX")
+
+    -- libisatty.so provides isatty() that always returns 1.
+    local libisatty_path = build_dir .. '../test/libisatty.so'
+
+    -- FreeBSD treates escaped chars differently compared to linux, thus the
+    -- escaping must be escaped too.
+    local tarantool_command
+    if (jit.os == 'BSD') then
+        tarantool_command = "local a = 0 \\\\\nfor i = 1, 10 do\\\na = a + i\\\nend \\\\\nprint(a)"
```

Comment:
As we discussed, you can just use double brackets here to pass all escaping as is.

Reply:
Fixed.